internal/util/telemetry/reporter_test.go

```diff
@@ -88,3 +93,62 @@ func TestNewReporterLock(t *testing.T) {
 		})
 	}
 }
+
+func TestReporterReport(t *testing.T) {
+	t.Parallel()
+
+	for name, tc := range map[string]struct {
+		f                *Flag
+		reporterResponse string
+		latestVersion    string
+		updateAvailable  bool
+	}{
+		"UpdateAvailable": {
+			f:                &Flag{v: pointer.ToBool(true)},
+			reporterResponse: `{"update_available": true, "latest_version": "0.3.4"}`,
+			latestVersion:    "0.3.4",
+			updateAvailable:  true,
+		},
+		"UpdateUnavailable": {
+			f:                &Flag{v: pointer.ToBool(true)},
+			reporterResponse: `{"update_available": false, "latest_version": "0.3.4"}`,
+			latestVersion:    "",
+			updateAvailable:  false,
+		},
+	} {
+		name, tc := name, tc
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			// use httptest.NewServer so http request on ts.URL returns tc.reporterResponse body
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusCreated)
```

Comment:
This is not a very accurate mock of becaon as beacon have a different logic for `GET` and `POST` methods - https://github.com/FerretDB/beacon/blob/e965bc7609649e31cca83783f95f32861afb38ac/internal/handler.go#L134 (the responses are very different for `GET` and `POST`.

Reply:
`r.report()` makes http `POST` request, so it does not call `GET` even though on API it's available. I added a check for request method type though :hugs:

Reply:
This one looks resolved, so I'm closing it